### PR TITLE
feat(shopify-checkout): support non GIDs for variantIds

### PR DESCRIPTION
**Pull request by mdarrik**
## WHY are these changes introduced?
Story [LS-1478](https://nacelle.atlassian.net/browse/LS-1478)

## WHAT is this pull request doing?

1. Adds a utility method for turning variantIds into valid shopify GIDs. This lets shopify checkout support v2 which currently strips off the `gid://` text before the id.
2. Uses that utility method in `cartItemsToCheckoutItems.ts` to convert each line item's variant Id to a `gid` if it wasn't already one.

## How to Test

1. Pull this PR locally
2. Run `npm run test:ci`

**Note** it'd probably be good to test this in an example project too to make sure that it works. Will update with instructions once I've gotten them.

### CODE REVIEW

**Review by alexcasche — APPROVED** [review]
LGTM

**Review by NWRichmond — CHANGES_REQUESTED** [review]
Looking good!

There are a couple of things that need attention before merging. I added some comments for little things, and I think the bigger thing is improving test coverage slightly.

The [coverage report](https://github.com/getnacelle/nacelle-js/pull/54#issuecomment-1021667778) shows a slip in coverage, due to `packages/shopify-checkout/src/utils/transformVariantIdToGid.ts`:

coverage report nacelle-js #54 c952536

When you look at https://github.com/getnacelle/nacelle-js/pull/54/files, the `jest-coverage-report-action` points out uncovered branches & statements:

branch is not covered

File: packages/shopify-checkout/src/utils/transformVariantIdToGid.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  const decodedVariantId = decodeBase64Id(variantId);
+
+  if (isPlainTextGlobalId(decodedVariantId)) {
+    return decodedVariantId;
+  }
+  // if it's a number, append the global id info to make it a valid id
+  if (isNumeric(decodedVariantId)) {
+    return idToVariantGid(decodedVariantId);
```
[the remark this answers is not in the document]
Comment (mdarrik) [reply packages/shopify-checkout/src/utils/transformVariantIdToGid.ts:23]: Yeah. That's essentially why I don't have a test for it. It doesn't hurt for us to have it, but we could also potentially get away without it.

[the remark this answers is not in the document]
Comment (mdarrik) [reply packages/shopify-checkout/src/utils/transformVariantIdToGid.ts]: Removed it just to make it a bit more robust.

**Comment by mdarrik** [comment]
> The coverage report shows a slip in coverage, due to packages/shopify-checkout/src/utils/transformVariantIdToGid.ts:


So the issue here is that it's got both node specific & browser specific code, and I'm not sure of a way to test both in this instance. It looks like it's testing the browser code, which is the one we mostly want tested, but I'm not sure how to make it test the node-specific code here.

**Comment by NWRichmond** [comment]
> > The coverage report shows a slip in coverage, due to packages/shopify-checkout/src/utils/transformVariantIdToGid.ts:
> 
> So the issue here is that it's got both node specific & browser specific code, and I'm not sure of a way to test both in this instance. It looks like it's testing the browser code, which is the one we mostly want tested, but I'm not sure how to make it test the node-specific code here.

@mdarrik I'd recommend checking out `packages/shopify-checkout/src/client/client-node.spec.ts` for inspiration.

**Comment by mdarrik** [comment]
@NWRichmond  I did look there. But I don't see anything there that would really make a difference? Unless `test-node` is a jest convenience to switch into the node environment, I'm not sure how to have `window === undefined` in the tests.

**Comment by NWRichmond** [comment]
> @NWRichmond I did look there. But I don't see anything there that would really make a difference? Unless `test-node` is a jest convenience to switch into the node environment, I'm not sure how to have `window === undefined` in the tests.

@mdarrik I had to refresh my memory, but the `@jest-environment node` [docblock](https://jestjs.io/docs/configuration#testenvironment-string) is the key here.

https://github.com/getnacelle/nacelle-js/blob/c952536750bd302247ce7c5f41984c7fceb17f9a/packages/shopify-checkout/src/client/client-node.spec.ts#L1-L3

**Comment by mdarrik** [comment]
Ah cool. That should be easy enough.

**Comment by mdarrik** [comment]
@NWRichmond added separate browser/node tests. Ended up with 100% coverage for the new file now.

**Review by NWRichmond — APPROVED** [review]
The coverage report is beautiful now!

Thanks you @mdarrik for this impactful PR 🙏

**Pull request merged**